### DISCUSSION TITLE
Interoperability layer for migration for Boost::Python to PyBind11

### DISCUSSION THREAD

**Discussion by duburcqa** [stack-of-tasks/pinocchio#1871] [Ideas]
I just wanted to point out [a very nice article](https://yyc.solvcon.net/en/latest/writing/2021/pyboost11/pyboost11.html) I found a year ago that is addressing a question I had for a very long time. How to setup a compatibility layer between PyBind11 and Boost::Python bindings. Hopefully this may allow to smooth the transition between both by enabling merging and actually deploying in production partial support of Pybind11. Maybe with some global Eigen-like switch (Matrix VS Array) it is incurs a major drop of performance.

**Comment by jcarpent** [comment]
Thanks @duburcqa for providing this reference.
I guess these convertes should be rather implemented within EigenPy for genericity.
It is also worth mentioning that there is no roadmap to switch to PyBind11, particularly due to serious performance issues for signature disambiguation at the PyBind11 level and also because we have developed an advanced support mechanism for any Scalar types which is central in Pinocchio 3x.
Yet, having strong interoperability between the two wrapping mechanisms is more than welcome.

PS: I will move this issue to a discussion.
